src/write.c

```diff
@@ -908,31 +910,27 @@ static avifResult avifEncoderAddImageItems(avifEncoder * encoder,
         gridItem->gridRows = gridRows;
         gridItem->gridWidth = gridWidth;
         gridItem->gridHeight = gridHeight;
-        *mainItemID = gridItem->id;
+        *topLevelItemID = gridItem->id;
     }
 
     for (uint32_t cellIndex = 0; cellIndex < cellCount; ++cellIndex) {
         avifEncoderItem * item = avifEncoderDataCreateItem(encoder->data, "av01", infeName, infeNameSize, cellIndex);
+        AVIF_CHECKERR(item, AVIF_RESULT_OUT_OF_MEMORY);
         item->codec = avifCodecCreate(encoder->codecChoice, AVIF_CODEC_FLAG_CAN_ENCODE);
-        if (!item->codec) {
-            // Just bail out early, we're not surviving this function without an encoder compiled in
-            return AVIF_RESULT_NO_CODEC_AVAILABLE;
-        }
+        // Just bail out early, we're not surviving this function without an encoder compiled in
+        AVIF_CHECKERR(item->codec, AVIF_RESULT_NO_CODEC_AVAILABLE);
```

Comment:
This comment was originally inside the if statement and was written for that location. The new code is equivalen to the following:

```
        // Just bail out early, we're not surviving this function without an encoder compiled in
        if (!item->codec) {
            return AVIF_RESULT_NO_CODEC_AVAILABLE;
        }
```

In the new location, the comment needs to describe the condition for bailing out early. For example,

```
        // Just bail out early if no encoder is available, we're not surviving this function without an encoder compiled in
        AVIF_CHECKERR(item->codec, AVIF_RESULT_NO_CODEC_AVAILABLE);
```

Or we can revert to the original code. We can also just delete the comment.

Reply:
Done in https://github.com/AOMediaCodec/libavif/pull/1299.